hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -75,17 +76,18 @@
   public HoodieLogFileReader(FileSystem fs, HoodieLogFile logFile, Schema readerSchema, int bufferSize,
                              boolean readBlockLazily, boolean reverseReader) throws IOException {
     FSDataInputStream fsDataInputStream = fs.open(logFile.getPath(), bufferSize);
-    if (FSUtils.isGCSInputStream(fsDataInputStream)) {
-      this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
+    boolean isGCSFileSystem = FSUtils.isGCSFileSystem(fs);
+    if (isGCSFileSystem) {
```

Comment:
I had a failure here running a job with a previous GCS detector, it would determine it to be GCS based on FSDataInputStream, but when it calls getWrappedStream().getWrappedStream() it ends up with GoogleHadoopFSDataInputStream that could not be cast to FSInputStream. So as a quick fix IF to this checking for FSInputStream first:
```java
    if (fsDataInputStream.getWrappedStream() instanceof FSInputStream) {
      this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
          new BufferedFSInputStream((FSInputStream) fsDataInputStream.getWrappedStream(), bufferSize)));
    } else if (FSUtils.isGCSInputStream(fsDataInputStream)) {
      this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
          new BufferedFSInputStream((FSInputStream) ((
              (FSDataInputStream) fsDataInputStream.getWrappedStream()).getWrappedStream()), bufferSize)));
    } else {
      // fsDataInputStream.getWrappedStream() maybe a BufferedFSInputStream
      // need to wrap in another BufferedFSInputStream the make bufferSize work?
      this.inputStream = fsDataInputStream;
    }
```

I would suggest you rearrange it here too as there is no need to do anything special if fsDataInputStream.getWrappedStream() is instanceof FSInputStream

Reply:
sure, will fix it.